File: app/src/main/java/com/demo/assignment/repository/logging/Logger.java
package com.demo.assignment.repository.logging;

import com.demo.assignment.util.AppUtils;

/**
 * The interface Logger.
 */
public interface Logger {
    /**
     * A {@link Logger} defaults output appropriate for the current platform.
     */
    Logger DEFAULT = message ->
            AppUtils.showLog(LoggingInterceptor.class.getSimpleName(), message);

    /**
     * Log.
     *
     * @param message the message
     */
    void log(String message);
}

File: app/src/main/java/com/digital/playground/repository/logging/Logger.kt
package com.digital.playground.repository.logging

import com.digital.playground.util.AppUtils.showLog

/**
 * The interface Logger.
 */
interface Logger {
    /**
     * Log.
     *
     * @param message the message
     */
    fun log(message: String)


    companion object {
        /**
         * A [Logger] defaults output appropriate for the current platform.
         */
        val DEFAULT: Logger = object : Logger {
            override fun log(message: String) {
                showLog(LoggingInterceptor::class.simpleName, message)
            }
        }
    }
}